MovingBlocks/Terasology
# Fixes #2382 Refresh Server Information in JoinGameScreen

### Contains
Fix for #2382

### How to test

- Start a headless run along with a normal run
- normal run instance should show GREEN information for Worlds and Modules labels in localhost under Custom tab
- kill the headless run
- Worlds and Modules information should still be GREEN, even after host is killed.
- Hit R key
- This should "unselect" the selected option on the left panel and clicking on localhost now would show RED "Connection Failed!" as expected.

The above should work for Listed/Online servers as well. ("Connection Failed!" goes back to "Requested." once the R key is pressed).

Pressing the R key unselects the selected option from `visibleList` and further clears `extInfo` which contains the hash table (between items on the left panel and server information) that needs to be refreshed.

### Outstanding before merging

- [x] Need to add a refresh button somewhere.
- [x] Probably need to refresh list on every new click in the left panel. (done in latest commit)
- [ ] Need to add translation for word "Refresh" in all other languages.
